src/main/java/com/aliyun/odps/jdbc/OdpsResultSet.java

```diff
@@ -344,24 +339,15 @@ public String getCursorName() throws SQLException {
       return null;
     } else if (obj instanceof java.util.Date) {
       return new java.sql.Date(((java.util.Date) obj).getTime());
+    } else if (obj instanceof String) {
+      SimpleDateFormat dateFormat = new SimpleDateFormat(JdbcColumn.ODPS_DATETIME_FORMAT);
+      try {
+        return new java.sql.Date(dateFormat.parse((String) obj).getTime());
+      } catch (ParseException e) {
+        throw new SQLException("can not parse to Date: " + obj, e);
+      }
     }
-
-    String strVal;
-    if (obj instanceof String) {
-      strVal = (String) obj;
-    } else if (obj instanceof byte[]) {
```

Comment:
@lyman since we change byte[]->String after getObject(), there is no need to check whether obj == byte[] (impossible).